File: poc/kotlin/Words.kt
import java.util.*;
import java.io.*;

object Words {

    @JvmStatic
    fun countOccurrences(line: String, word: String): Int {
        return line.split(" ").count { w -> w == word }
    }

    @JvmStatic
    fun main(args: Array<String>) {

        val fileName = args[0];
        val word = args[1];

        try {
            var count = 0
            File(fileName).forEachLine { line ->
               count += countOccurrences(line, word)
            }
            
            println("Word was found $count times")

        } catch (exception: FileNotFoundException) {
            println("Error trying to read the file $fileName")
        }
    } 
}

File: poc/go/words.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	fileName := os.Args[1]
	wordToCount := os.Args[2]

	file, err := os.Open(fileName)
	if err != nil {
		panic(err)
	}

	defer file.Close()

	scanner := bufio.NewScanner(file)
	count := 0
	for scanner.Scan() {
		count += countOcurrences(scanner.Text(), wordToCount)
	}

	fmt.Printf("Word was found %d time\n", count)

	if err := scanner.Err(); err != nil {
		fmt.Println("error reading the file")
	}
}

func countOcurrences(line, w string) int {
	words := strings.Split(line, " ")
	count := 0
	for _, word := range words {
		if word == w {
			count += 1
		}
	}

	return count
}
